### DISCUSSION TITLE
Environment variables not working in Linux Ubuntu, but works in Windows

### DISCUSSION THREAD

**Discussion by bso-oo** [prisma/prisma#19107] [Q&A]
### Bug description

I have set up the DATABASE_URL environment variable using other variables like this:
`example
USER=postgres
PASSWORD=avcd10!
HOST=x.x,10.9
PORT=5432
DATABASE=xxx_dev
`
`DATABASE_URL=postgresql://${USER}:${PASSWORD}@${HOST}:${PORT}/${DATABASE}?schema=public`

`datasource db {
  provider   = "postgresql"
  url        = env("DATABASE_URL")
  extensions = [pgcrypto]
}`

It doesn't work in Linux Ubuntu, but it works fine in my Windows local environment. When I removed the variables and hard-coded the values directly in the DATABASE_URL, it worked in Linux Ubuntu.

Why are the environment variables not working in Linux Ubuntu, while they work in Windows? Is there any difference in handling environment variables between the two operating systems?

### How to reproduce

npx prisma migrate dev
Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma
Datasource "db": PostgreSQL database "xxx_dev", schema "public" at "localhost:5432"

Error: P1000: Authentication failed against database server at `localhost`, the provided database credentials for `(not available)` are not valid.

Please make sure to provide valid database credentials for the database server at `localhost`.

### Expected behavior

 I removed the variables and hard-coded the values directly in the DATABASE_URL
 
 `DATABASE_URL=postgresql://postgres:avcd10!@localhost:5432/xxx_dev?schema=public
`

### Prisma information

```prisma
// Add your schema.prisma
```

```ts
// Add your code using Prisma Client
generator client {
  provider        = "prisma-client-js"
  previewFeatures = ["postgresqlExtensions"]
}
```

### Environment & setup

- OS: 
- Database: 
- Node.js version: 
OS: Ubuntu
Database: Postgresql
node versuin: 16.20.0

### Prisma Version

```
prisma                  : 4.13.0
@prisma/client          : 4.13.0
Current platform        : debian-openssl-1.0.x
Query Engine (Node-API) : libquery-engine 1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a (at node_modules/@prisma/engines/libquery_engine-debian-openssl-1.0.x.so.node)
Migration Engine        : migration-engine-cli 1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a (at node_modules/@prisma/engines/migration-engine-debian-openssl-1.0.x)
Format Wasm             : @prisma/prisma-fmt-wasm 4.13.0-50.1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a
Default Engines Hash    : 1e7af066ee9cb95cf3a403c78d9aab3e6b04f37a
Studio                  : 0.484.0
Preview Features        : postgresqlExtensions

```

**Comment by nurul3101** [comment]
Hello @bso-oo 👋 

Once you set the environment variable in the Linux machine, can you check if the value of `DATABASE_URL` is correct at runtime?
You can use the dotenv package to load the environment variable as specified here: https://www.prisma.io/docs/guides/development-environment/environment-variables#using-environment-variables-in-your-code

**Reply by nurul3101** [reply]
I am checking to see if you had a chance to look at my last response.

**Reply by nurul3101** [reply]
If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏
